src/pprof/ddprof_pprof.cc

```diff
@@ -136,17 +123,16 @@ static void write_line(const ddprof::Symbol &symbol,
 // Assumption of API is that sample is valid in a single type
 DDRes pprof_aggregate(const UnwindOutput *uw_output,
                       const SymbolHdr *symbol_hdr, uint64_t value,
-                      int watcher_idx, DDProfPProf *pprof) {
+                      const PerfWatcher *watcher, DDProfPProf *pprof) {
 
   const ddprof::SymbolTable &symbol_table = symbol_hdr->_symbol_table;
   const ddprof::MapInfoTable &mapinfo_table = symbol_hdr->_mapinfo_table;
   ddprof_ffi_Profile *profile = pprof->_profile;
 
-  int64_t values[MAX_TYPE_WATCHER + 1] = {0};
-  // Counted a single time
-  values[0] = 1;
-  // Add a value to the watcher we are sampling, leave others zeroed
-  values[watcher_idx + 1] = value;
+  int64_t values[DDPROF_PWT_LENGTH] = {0};
+  values[watcher->profile_id] = value;
```

Comment:
There is a off by one error here since `profile_id` starts from 1 (0 is `DDPROF_PWT_NOCOUNT`)

Reply:
Good find.  I sloppily added that change without thorough testing and assumed it was a backend issue.  I think this is fixed now that I've changed it so the sample types are dynamically populated and referred to through a permutation vector between the static IDs and the used IDs